### PR TITLE
Add imprint API endpoint, fixes #80

### LINKED ISSUE #80: API: city endpoints needs imprint
The city endpoint needs to define which pages are the imprint. This can be set with the slug of the imprint page.

https://github.com/Integreat/cms/wiki/REST-APIv3-Documentation#pageseventslocationsdisclaimer

~Also consider the discussion in https://github.com/Integreat/cms/issues/916~

### PR DESCRIPTION
**Pull request by svenseeberg**
* Add imprint API endpoint
* Make trailing slashes optional for all API endpoints
* Merge offer/extra and offers/extras URL patterns

### CODE REVIEW

File: src/api/urls.py [hunk 1]
```diff
@@ -22,67 +23,60 @@
...
+    url(r"regions/hidden/?$", hiddenregions),
+    url(r"regions/pushnew/?$", pushnew),
     url(
         r"(?P<region_slug>[-\w]+)/",
         include(
             [
-                url(r"languages$", languages),
-                url(r"offers$", offers),
+                url(r"languages/?$", languages),
+                url(r"offers/?$", offers),
```
Comment (timoludwig) [inline src/api/urls.py:35]:
```suggestion
                url(r"(extras|offers)/?$", offers),
```
Should we add both urls here as well?

File: src/api/urls.py [hunk 2]
```diff
@@ -22,67 +23,60 @@
...
                 url(
-                    r"(?P<language_code>[-\w]+)/feedback/events$",
+                    r"(?P<language_code>[-\w]+)/feedback/events/?$",
                     event_list_feedback.event_list_feedback,
                 ),
-                url(r"(?P<language_code>[-\w]+)/pages/$", pages),
-                url(r"(?P<language_code>[-\w]+)/offers/$", offers),
-                url(r"(?P<language_code>[-\w]+)/page/$", single_page),
+                url(r"(?P<language_code>[-\w]+)/pages/?$", pages),
+                url(r"(?P<language_code>[-\w]+)/offers/?$", offers),
```
Comment (timoludwig) [inline src/api/urls.py:77]:
```suggestion
                url(r"(?P<language_code>[-\w]+)/(extras|offers)/?$", offers),
```
Should we add both urls here as well?

File: src/api/v3/imprint.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    return {
+        "id": imprint_translation.id,
+        "url": imprint_translation.permalink,
+        "title": imprint_translation.title,
+        "modified_gmt": imprint_translation.last_updated,
+        "excerpt": imprint_translation.text,
+        "content": imprint_translation.text,
+        "parent": None,
+        "available_languages": imprint_translation.available_languages,
+        "thumbnail": None,
```
Comment (timoludwig) [inline src/api/v3/imprint.py:28]:
```suggestion
        "thumbnail": imprint_translation.page.icon,
```
The `ImprintPage` model also has an `icon` attribute. Should we use it here?

File: src/api/urls.py (hunk 2 above)
Comment (svenseeberg) [reply src/api/urls.py:77]: yes

File: src/api/v3/imprint.py (hunk 3 above)
Comment (svenseeberg) [reply src/api/v3/imprint.py:28]: yes

File: src/api/urls.py (hunk 1 above)
Comment (svenseeberg) [reply src/api/urls.py:35]: yes

**Review by timoludwig — APPROVED** [review]
Looks good to me! :+1:

**Pull request merged**